projects/media-viewer/src/lib/toolbar/media-viewer-toolbar/media-viewer-toolbar.component.ts

```diff
@@ -0,0 +1,281 @@
+    new ConnectionPositionPair(
+      {
+        originX: 'end',
+        originY: 'bottom'
+      },
+      {
+        overlayX: 'end',
+        overlayY: 'top'
+      },
+      0,
+      3)
+  ];
+
+  public constructor(
+    public readonly toolbarEvents: ToolbarEventService,
+    public readonly toolbarButtons: ToolbarButtonVisibilityService,
+    private readonly cdr: ChangeDetectorRef,
+    private readonly numberHelper: NumberHelperService
+  ) {
+  }
+
+  public ngOnInit() {
+    this.subscriptions.push(
+      this.toolbarEvents.setCurrentPageSubject.subscribe(pageNumber => this.setCurrentPage(pageNumber)),
+      this.toolbarEvents.setCurrentPageInputValueSubject.subscribe(pageNumber => this.pageNumber = pageNumber),
+      this.toolbarEvents.getPageCount().subscribe(count => this.pageCount = count),
+      this.toolbarEvents.icp.enabled.subscribe(enabled => {
+        this.icpEnabled = enabled;
+        if (this.icpEnabled) {
+          this.toolbarEvents.toggleCommentsPanel(!enabled);
+          this.toolbarEvents.subToolbarHidden.next(true);
+          this.toolbarEvents.sidebarOpen.next(false);
+        }
+      }),
+      this.toolbarEvents.redactionMode.subscribe(enabled => {
+        this.redactionEnabled = enabled;
+      }),
+
+      this.toolbarEvents.commentsPanelVisible.subscribe(toggle => this.showCommentsPanel = toggle)
+    );
+  }
+
+  public ngOnDestroy(): void {
+    for (const subscription of this.subscriptions) {
+      subscription.unsubscribe();
+    }
+  }
+
+  public ngAfterViewInit() {
+    let totalSpace = 0;
+
+    // Get all toolbar elements
+    const toolbarElements: HTMLElement[] = Array.prototype.slice.call(this.mvToolbarMain.nativeElement.children);
+
+    // Set a flag for each element's visibility and push to array of button states
+    for (const element of toolbarElements) {
+      this.toolbarButtonVisibilityStates.push({ id: element.id, showOnToolbar: true } as ButtonState);
+    }
+
+    // Trigger change detection (necessary to ensure all toolbar elements are rendered fully - and have a size - at this stage)
+    this.cdr.detectChanges();
+
+    for (const element of toolbarElements) {
+      this.totalButtonSpace += element.getBoundingClientRect().width;
+    }
+
+    // add the main toolbar left margin to the available space
+    this.totalButtonSpace = this.totalButtonSpace + this.mvToolbarMain.nativeElement.getBoundingClientRect().left;
+
+    // Loop over all elements and set sum of widths for each toolbar element
+    for (const element of toolbarElements) {
+      // Use bounding client rectangle width for accuracy
+      totalSpace += element.getBoundingClientRect().width;
+      this.breakWidths.push(totalSpace);
+    }
+    // Call calculation method
+    this.checkCalculation();
+  }
+
+  @HostListener('window:resize', [])
+  public onResize() {
+    this.cdr.detectChanges();
+    // Call calculation method
+    this.checkCalculation();
+  }
+
+  private checkCalculation() {
```

Comment:
this is way too complex for what it's doing

Reply:
Related to code complexity. Potential for additional technical debt if a different approach is adopted.